### ISSUE TITLE
Large performance discrepancy between validation and test splits in weak label evaluation

### ISSUE THREAD

**Issue by Kasi21** [medhaaga/BotswanaML#34]
When evaluating the FixMatch models, we're seeing substantial differences in the confusion matrices between the val and test splits across several behaviours, particularly for Feeding and Moving. In the validation split, performance appears strong and relatively balanced, whereas in the test split there is a noticeable drop in performance for Feeding.

At this stage, no calibration or threshold tuning is being performed using the validation set, so I would have expected the validation and test results to be more similar in magnitude.

Could you take a look and investigate why this is happening 

<img width="1603" height="409" alt="Image" src="https://github.com/user-attachments/assets/1c6ef413-1a38-4457-9b26-ba7e258969cc" />

<img width="1592" height="418" alt="Image" src="https://github.com/user-attachments/assets/dfa0bde0-16ce-42f2-9ea8-d9c7bf03e7a5" />

**Comment by medhaaga** [comment]
Primary model selection during training is based on performance on the validation set. Over the course of training (in this case, 200 epochs), model performance can fluctuate substantially. It is therefore standard practice to track and retain the checkpoint that achieves the lowest validation loss, rather than automatically selecting the model from the final (200th) training epoch.
As a result, it is expected that model performance on the validation splits, across both labeled source and labeled target data, will be stronger than performance on the held-out test split.
